addons/payment/models/payment_acquirer.py

```diff
@@ -971,11 +971,11 @@ def form_feedback(self, data, acquirer_name):
     # --------------------------------------------------
 
     def s2s_do_transaction(self, **kwargs):
-        custom_method_name = '%s_s2s_do_transaction' % self.acquirer_id.provider
```

Comment:
I would definitely prefer a call to `ensure_one()` here, instead of looping. Here's why:
- I prefer failing fast when dealing with payment providers. With ensure_one, it fails before contacting any external service (no orphan payment). With a loop, if any payment in the set fails, the whole transaction is rollbacked (even though some payments may actually have gone through)
- You loose the return value in this multi version you propose

Reply:
Hello @dbo-odoo 

Then we should make a loop here https://github.com/odoo/odoo/blob/master/addons/payment/models/account_payment.py#L114
What do you think?

Thanks :)

Reply:
You're right, you'll need to do that. If it wasn't for the lost `return` value I wouldn't bother and merge with the loop; I still insist on the `ensure_one` though.

You can change the call in accounting to a loop and update the PR :)

thx!